### PR TITLE
LATCH-335 Update component Tag with icon

### PR DESCRIPTION
**Pull request by jaranaz**
Update the existent Tag component for SwiftUI to use icons.

Described in Jira task [LATCH-335](https://jira.tid.es/browse/LATCH-335)

### CODE REVIEW

**Comment by salavert** [comment]
Remember to follow the guide for semantic releases https://github.com/Telefonica/mistica-ios/blob/main/CONTRIBUTING.md

File: Sources/MisticaSwiftUI/Components/Tag/README.md [hunk 1]
```diff
@@ -1,18 +1,30 @@
 # Tag
 
-This view consists of a label to display text with a defined style.
```
Comment (salavert) [inline Sources/MisticaSwiftUI/Components/Tag/README.md:3]: Preview here:
https://github.com/Telefonica/mistica-ios/blob/LATCH-335-update-mistica-tag-with-icon/Sources/MisticaSwiftUI/Components/Tag/README.md

**Review by salavert — APPROVED** [review]

File: Sources/MisticaSwiftUI/Components/Tag/Tag.swift [hunk 2]
```diff
@@ -28,25 +30,43 @@ public struct Tag: View {
 
     private let style: Style
     private let text: String
+    private let icon: Image?
     private var isInverse = false
+    private var textAccessibilityIdentifier: String?
+    private var textAccessibilityIdentifierIcon: String? {
```
Comment (cbarcom) [inline Sources/MisticaSwiftUI/Components/Tag/Tag.swift:36]: should this be iconAccessibilityIdentifier?

File: Sources/MisticaSwiftUI/Components/Tag/Tag.swift [hunk 3]
```diff
@@ -28,25 +30,43 @@ public struct Tag: View {
 
     private let style: Style
     private let text: String
+    private let icon: Image?
     private var isInverse = false
+    private var textAccessibilityIdentifier: String?
+    private var iconAccessibilityIdentifier: String? {
```
Comment (idenjoe) [inline Sources/MisticaSwiftUI/Components/Tag/Tag.swift:36]: Maybe we could use a guard here like this:
```swift
private var iconAccessibilityIdentifier: String? {
       guard let textAccessibilityIdentifier = textAccessibilityIdentifier else {
            returni nil
      }

      return textAccessibilityIdentifier + "-icon"
}
```

**Review by idenjoe — APPROVED** [review]

File: Sources/MisticaSwiftUI/Components/Tag/Tag.swift (hunk 3 above)
Comment (cbarcom) [reply Sources/MisticaSwiftUI/Components/Tag/Tag.swift:36]: Has this been agreed with Android?

**Review by salavert — APPROVED** [review]

File: Sources/MisticaSwiftUI/Components/Tag/Tag.swift (hunk 3 above)
Comment (jaranaz) [reply Sources/MisticaSwiftUI/Components/Tag/Tag.swift:36]: Talked to Android team, they confirm they're not using this kind of identifiers in Mistica components.

In this particular PR, the sufix "-icon" for the Tag icon has been replicated from the UIKit Tag component.

Comment (cbarcom) [reply Sources/MisticaSwiftUI/Components/Tag/Tag.swift:36]: weird, AFAIK, they have default values for accessibility identifiers and we have talked about they give the option to customize them. In Smartwifi, they are very useful for automated tests and are being used.

Although if they haven't defined for this component, we have advantage on this and can choose whatever we think it fits

**Review by cbarcom — APPROVED** [review]

**Pull request merged**